### DISCUSSION TITLE
Shapes

### DISCUSSION THREAD

**Discussion by YRobles** [ShapeCrawler/ShapeCrawler#442] [Q&A]
Hi!

I'm wondered if is there a way to create or clone shapes in a slide? I have several circles (like a traffic lights) that I need duplicate several times (one for each row of a dynamic table).

In addition, I have again the issue with the exception when filling a shape. 
I have this shape called Presupuesto0

![image](https://user-images.githubusercontent.com/4245483/214514706-6558a9a7-7c87-499c-8dd5-12fcc9bf63f1.png)

And when I tried to fill with a color, I'm getting this exception:

![image](https://user-images.githubusercontent.com/4245483/214514611-7cf67de1-cde3-4bcd-8463-6cad121dfe4b.png)

I guess that when the shape is inside another shape or table, it works fine, but when you create the shape outside any other shape, it throw doesn't have parent exception... but I don't know why.. 

Thanks in advanced and for the marvelous work you do!

**Comment by AdamShakhabov** [comment]
Hello.

No, it does not support shape clone. However, it is a good idea, and we will do it (#444)

Currently, there are methods to create only a new Rectangle or Rounded Rectangle shapes, for example how to add a new Rectangle:

```c#
var pres = SCPresentation.Create();
var shapeCollection = pres.Slides[0].Shapes;
var addedShape = shapeCollection.AutoShapes.AddRectangle(x: 50, y: 60, w: 100, h: 70);
addedShape.TextFrame!.Text = "Hello World!";
pres.SaveAs("my_pres.pptx");
```
---

_Regard Shape Filling, I'm investigating this..._

**Reply by AdamShakhabov** [reply]
Here is the fix for color filling [**0.41.3-beta.1**](https://github.com/ShapeCrawler/ShapeCrawler/files/10527253/ShapeCrawler.0.41.3-beta.1.nupkg.zip).

Let me know if it doesn't work.

**Reply by YRobles** [reply]
Hi, thanks for the work. I have tried it just now, but it doesn't work. Now Fill method doesn't throw an exception but it doesn't change the color. It don't happen with all shapes, I have several ppt and it works fine with the same shapes that before did, but not with the shapes that I have the problem. I think it works with shapes that are inside other objects, and if the shape doesn't have "parent" it doesn't do anything. I'm going to make some tests about different kind of shapes... Thanks for all.

**Reply by AdamShakhabov** [reply]
It should work in the last published version.

**Reply by YRobles** [reply]
Thanks for all. It works!
